test/test_proxy_tensor.py

```diff
@@ -110,6 +111,16 @@ def f_backward(x):
             assert inp.grad is None
             torch.testing.assert_close(traced_graph_out, f(inp))
 
+    def test_inplace_metadata(self):
+        def f(x):
+            x.unsqueeze_(-1)
+            assert x.shape[-1] == 1
+            return x
+
+        inps = [torch.randn(5)]
+        fx_f = make_fx(f)(*inps)
+        self.assertEqual(fx_f(*inps), f(*inps))
```

Comment:
I'm confused: I would expect this to error out. if f(*inps) mutates the input, then fx_f(*inps) should also mutate the input, and if they both mutate the input then the result should have different shapes.

Does self.assertEqual not check for shape equality?

Reply:
Apparently not.